usr.sbin/etcupdate/etcupdate.sh

```diff
@@ -184,14 +184,24 @@ always_install()
 # $1 - directory to store new tree in
 build_tree()
 (
-	local destdir dir file make
+	local destdir dir file make autogenfiles metatmp
 
 	make="$MAKE_CMD $MAKE_OPTIONS -DNO_FILEMON"
 
+	if [ -n "$noroot" ]; then
```

Comment:
#Note that $1 is already a "temporary" tree created with mktemp -d when this is called, so you could possibly use $1/METALOG.rm or some such as metatmp without having to deal with cleanup on error (would just need to purge it at the end of the function) or mktemp.

Reply:
Yeah, I didn't really want to assume anything about how the function is called

Reply:
It is an intended API assumption as the function doesn't cleanup $1 on error but relies on the caller to handle that, but either way is fine.